cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -155,25 +180,30 @@ private AccessToken acquireAccessToken(ClientConfig clientConfig, ConnectionConf
     }
 
     try {
-      AccessToken savedAccessToken = getSavedAccessToken(connectionInfo.getHostname());
-      checkConnection(clientConfig, connectionInfo, savedAccessToken);
-      return savedAccessToken;
+      UserAccessToken savedToken = getSavedAccessToken(connectionInfo.getHostname());
+      connectionInfo = new AuthenticatedConnectionConfig(connectionInfo, savedToken.getUsername());
+      clientConfig.setConnectionConfig(connectionInfo);
+      checkConnection(clientConfig, savedToken.getAccessToken());
+      return savedToken.getAccessToken();
     } catch (UnauthorizedException ignored) {
       // access token invalid - fall through to try acquiring token manually
     }
 
-    return getNewAccessToken(connectionInfo, output, debug);
+    return getNewAccessToken(clientConfig, output, debug);
```

Comment:
Actually, the new connection is set here: https://github.com/caskdata/cdap/blob/feature/show-username/cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java#L146. You don't need to set it again in this method. Also this method is only intended to return a new access token that is obtained by asking the user for credentials, so it doesn't need to modify anything.